web/packages/teleport/src/Roles/RoleEditor/standardmodel.ts

```diff
@@ -409,9 +473,14 @@ function roleConditionsToAccessSpecs(conditions: RoleConditions): {
     });
   }
 
+  const { model: rulesModel, requiresReset: rulesRequireReset } =
+    rulesToModel(rules);
+
   return {
     accessSpecs,
-    requiresReset: !isEmpty(rest),
+    rules: rulesModel,
+    requiresReset:
+      kubernetesResourcesRequireReset || rulesRequireReset || !isEmpty(rest),
```

Comment:
It might be a good idea to explain here why `!isEmpty(rest)` means that a reset is required and perhaps rename `rest` to `conditionsUnsupportedByEditor` or something.